### PR TITLE
allow `genetic_relatedness_vector` to do windows that don't span the whole genome

### PR DESCRIPTION
**Pull request by petrelharp**
Pretty straightforward; and this allows parallelization across windows.

*Edit:* I've added into this the ability to pass in a different set of nodes (added here since it's on top of these changes).

### CODE REVIEW

File: c/tskit/trees.c [hunk 1]
```diff
@@ -9977,6 +9984,20 @@ tsk_matvec_calculator_init(tsk_matvec_calculator_t *self, const tsk_treeseq_t *t
     tsk_memset(result, 0, num_windows * num_samples * num_weights * sizeof(*result));
     tsk_memset(self->parent, TSK_NULL, num_nodes * sizeof(*self->parent));
 
+    ret = tsk_tree_position_init(&self->tree_pos, ts, 0);
+    if (ret != 0) {
+        goto out;
```
Comment (petrelharp) [inline c/tskit/trees.c:9989]: this function always returns 0 currently

File: c/tskit/trees.c [hunk 2]
```diff
@@ -9977,6 +9984,20 @@ tsk_matvec_calculator_init(tsk_matvec_calculator_t *self, const tsk_treeseq_t *t
...
+    }
+    /* seek to the first window */
+    index = (tsk_id_t) tsk_search_sorted(breakpoints, num_trees + 1, windows[0]);
+    if (breakpoints[index] > windows[0]) {
+        index--;
+    }
+    ret = tsk_tree_position_seek_forward(&self->tree_pos, index);
+    if (ret != 0) {
+        goto out;
```
Comment (petrelharp) [inline c/tskit/trees.c:9998]: same here

**Comment by petrelharp** [comment]
I think this is ready to go - the only uncovered lines are not cover-able.

**Comment by petrelharp** [comment]
I've also put in the code here to let this work with arbitrary nodes. To see those changes separately look at the commits.

**Comment by petrelharp** [comment]
This looks good to go! (One test will need to be re-run but it failed for other reasons.)

**Review by jeromekelleher — APPROVED** [review]
LGTM! I haven't gone through the fine details, but generally look good.

**Pull request merged**